src/rust/ensogl/lib/core/src/data/mix.rs

```diff
@@ -0,0 +1,102 @@
+//! Abstraction for values that can be mixed together. For numbers this is just a simple linear
+//! interpolation. For other types it can get more complex. For example, mixing colors requires
+//! to convert them to a linear space first, do the interpolation and then performing a conversion
+//! back.
+
+use crate::prelude::*;
+
+
+
+// =============
+// === Space ===
+// =============
+
+/// Strongly typed value representation in the mix space.
+#[allow(missing_docs)]
+pub struct Space<T:Mixable> {
+    pub value : Repr<T>
+}
```

Comment:
I this very case I would prefer `MixSpace` or `MixingSpace` - I'm not sure why, but - especially in this module - the `Space` alone does not look well. Perhaps the `Space` alone sounds to me like "some set of values" instead of single value in a concrete space.

Reply:
But we are using this thing outside of the module as `mix::Space`, I don't want to expose it as `mix::MixSpace`. The name of the module matters.